### PR TITLE
Proximity sensors tripped by ghosts

### PR DESCRIPTION
**Pull request by Arivias**
[BUG]
## About the PR 
Prevents proximity sensors from being triggered by ghosts.

## Why's this needed? 
https://github.com/goonstation/goonstation/issues/2239

### CODE REVIEW

File: code/obj/item/assembly/misc_assemblies.dm [hunk 1]
```diff
@@ -247,7 +247,7 @@ Contains:
 
 /obj/item/assembly/prox_ignite/HasProximity(atom/movable/AM as mob|obj)
 
-	if (istype(AM, /obj/projectile))
+	if (istype(AM,/mob/dead) || istype(AM, /obj/projectile))
```
Comment (ThePotato97) [inline code/obj/item/assembly/misc_assemblies.dm:250]:
```suggestion
	if (isobserver(AM) || istype(AM, /obj/projectile))
```

File: code/obj/item/device/prox_sensor.dm [hunk 2]
```diff
@@ -69,7 +69,7 @@
 	return
 
 /obj/item/device/prox_sensor/HasProximity(atom/movable/AM as mob|obj)
-	if (istype(AM, /obj/projectile) || AM.invisibility > 2)
+	if (istype(AM, /mob/dead) || istype(AM, /obj/projectile) || AM.invisibility > 2)
```
Comment (ThePotato97) [inline code/obj/item/device/prox_sensor.dm:72]:
```suggestion
	if (isobserver(AM) || istype(AM, /obj/projectile) || AM.invisibility > 2)
```

**Review by ThePotato97 — CHANGES_REQUESTED** [review]
Use isobserver(x) instead of istype.

**Comment by ThePotato97** [comment]
Shouldn't `AM.invisibility > 2` be checking if they're a ghost/wraith/anything invisible?

**Comment by ThePotato97** [comment]
Observers don't set off the proximity sensor, i just tested it but wraiths do fix would be editing
https://github.com/goonstation/goonstation/blob/146d2e57479d7e0bd3f27325f7d72e011155b8f2/code/modules/networks/computer3/mainframe2/misc_terms.dm#L2821
`if(isobserver(AM) || isintangible(AM) || AM.invisibility > 2) return`

**Review by ThePotato97 — APPROVED** [review]

File: code/obj/item/assembly/misc_assemblies.dm [hunk 3]
```diff
@@ -247,7 +247,7 @@ Contains:
 
 /obj/item/assembly/prox_ignite/HasProximity(atom/movable/AM as mob|obj)
 
-	if (istype(AM, /obj/projectile))
+	if (isobserver(AM) || istype(AM, /obj/projectile))
```
Comment (ZeWaka) [inline code/obj/item/assembly/misc_assemblies.dm:250]:
```suggestion
	if (isobserver(AM) || iswraith(AM) || isintangible(AM) || istype(AM, /obj/projectile))
```

File: code/obj/item/device/prox_sensor.dm [hunk 4]
```diff
@@ -69,7 +69,7 @@
 	return
 
 /obj/item/device/prox_sensor/HasProximity(atom/movable/AM as mob|obj)
-	if (istype(AM, /obj/projectile) || AM.invisibility > 2)
+	if (isobserver(AM) || istype(AM, /obj/projectile) || AM.invisibility > 2)
```
Comment (ZeWaka) [inline code/obj/item/device/prox_sensor.dm:72]:
```suggestion
	if (isobserver(AM) || iswraith(AM) || isintangible(AM) || istype(AM, /obj/projectile)) || AM.invisibility > 2)
```
humans can be invisible too, which is why we check invisibility

**Review by ZeWaka — CHANGES_REQUESTED** [review]
intangible and wraiths

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**